### PR TITLE
Add spell check github action for the repo

### PR DESCRIPTION
**Pull request by balakreddy**
Fix some of the typos and added new github action
which runs the spell check as mentioned in https://github.com/Azure/GuestProxyAgent/issues/55#issue-2314136476 
on the pull requests.

### CODE REVIEW

**Comment by balakreddy** [comment]
@microsoft-github-policy-service agree company="Microsoft"

**Comment by balakreddy** [comment]
Hello @ZhidongPeng and @shahneerali 
Can you review this when you have time.

Thanks in advance

File: .github/actions/spelling/expect.txt [hunk 1]
```diff
@@ -0,0 +1,339 @@
+aab
+AAFFBB
+abe
+addrpair
+aduit
```
Comment (ZhidongPeng) [inline .github/actions/spelling/expect.txt]: this must be a typo, could you please help update this list? I assume it is auto-generated.

File: ebpf/redirect.bpf.c [hunk 2]
```diff
@@ -125,19 +125,16 @@ authorize_v4(bpf_sock_addr_t *ctx)
...
-        // }
-        // else
-        // {
-        //     ctx->user_ip4 = ctx->msg_src_ip4;
-        //     bpf_printk("Local/source ip is set, redirect to source ip:%u.", ctx->user_ip4);
-        // }
-
-        bpf_printk("redirecting to destination loopback ip.");
-        ctx->user_ip4 = policy->destination_ip.ipv4;
+        if (ctx->msg_src_ip4 == 0)
```
Comment (ZhidongPeng) [inline ebpf/redirect.bpf.c]: seems like the auto-merge revert my changes.

**Review by ZhidongPeng — CHANGES_REQUESTED** [review]

File: ebpf/redirect.bpf.c (hunk 2 above)
Comment (ZhidongPeng) [reply ebpf/redirect.bpf.c]: https://github.com/Azure/GuestProxyAgent/commit/13dc191ddf1e911e376485655d5fa762a2acb9f1#diff-c2fd1f1dcf93f19078fc1edf1937906ab6e5315a93d86b7283edcf344fbadcff

Comment (balakreddy) [reply ebpf/redirect.bpf.c]: Ack, will revert these changes.

**Comment by ZhidongPeng** [comment]
Thanks for your inputs. replacing it with https://github.com/Azure/GuestProxyAgent/pull/136

**Pull request closed without merging**